Do not unquote integers with a starting zero

phaer opened the pull request:
Integers with leading zeros are invalid in JSON, so they should probably stay quoted in. (They would be valid Javascript, but JSON does not support octal notation as stated on http://json.org.

Fixes #389

solarkennedy commented:
Please add a test case for this. If we don't, then this particular functionality will be lost as consul_sorted_json changes.

solarkennedy commented:
@phaer does this also fix #283 ?

phaer commented:
Just added tests and fixed the regex to still unquote for the case where the number is just '0'. As far as I understand, it does *not* fix #283 which looks for a way to keep valid integers quoted. This PR is only about (invalid) integers with a leading zero.
